### ISSUE TITLE
Tpetra: Make CrsGraph & CrsMatrix fillComplete check domain & range Map one-to-one-ness in debug mode

### ISSUE THREAD

**Issue by mhoemmen** [trilinos/Trilinos#1925]
@trilinos/tpetra  
See https://github.com/NaluCFD/Nalu/issues/211 for an example of a common user error, namely not passing in domain and range Maps to fillComplete when the row Map is overlapping (i.e., not one to one).

Tpetra has the `TPETRA_DEBUG` environment variable now; if you set it to 1, you'll get more debug-mode checks.  We can exploit this to help users diagnose their incorrect usage of Tpetra.  In particular, we can do the following:

  - Make the version of fillComplete that takes no Maps check that the row Map is one to one
  - Make the version of fillComplete that takes the domain and range Maps check that these Maps are one to one

We should only do these checks in debug mode, since they may be expensive (they may require extra MPI communication).

**Comment by mhoemmen** [comment]
Pull Request #1930 fixes this issue, but causes some downstream tests to fail.  Users are welcome to try that pull request for debugging.

**Comment by csiefer2** [comment]
@jhux2 is playing in a box related to this, namely checking valid inputs to the Tpetra::Import constructor (again DEBUG only)

**Comment by mhoemmen** [comment]
ping

**Comment by mhoemmen** [comment]
@csiefer2 @jhux2 @aprokop

**Comment by aprokop** [comment]
@mhoemmen Sorry, I'm missing some context. What would you like us to do?

**Comment by jhux2** [comment]
@aprokop @mhoemmen is waiting on me, I think.

**Comment by mhoemmen** [comment]
@jhux2 just an innocent ping :)

**Comment by jhux2** [comment]
:ping_pong:

**Comment by mhoemmen** [comment]
not gonna let this autoclose

**Comment by mhoemmen** [comment]
I can't add a "WONTFIX" label; y'all ought to do that : - )
